src/app/tabs/registrar/operations.js

```diff
@@ -330,5 +404,51 @@ export const checkIfInProgress = domain => (dispatch) => {
   };
 
   // if rLogin does not exist, start with the modal:
-  return (window.rLogin) ? callback() : dispatch(start(callback));
+  return window.rLogin ? callback() : dispatch(start(callback));
+};
+
+export const checkIfRequiresCommitment = domain => async (dispatch, getState) => {
+  dispatch(requestIsCommitmentRequired());
+
+  if (!partnerConfigurationAddress) {
+    return dispatch(receiveIsCommitmentRequired(true));
+  }
+
+  const web3 = new Web3(rskNode);
+  const partnerConfiguration = new web3.eth.Contract(
+    partnerConfigurationAbi,
+    partnerConfigurationAddress,
+  );
+
+  try {
+    const minCommitmentAge = await partnerConfiguration.methods
+      .getMinCommitmentAge()
+      .call();
+
+    const isCommitmentRequired = minCommitmentAge !== '0';
+
+    if (!isCommitmentRequired) {
+      const randomBytes = window.crypto.getRandomValues(new Uint8Array(32));
+      const strSalt = Array.from(randomBytes)
+        .map(byte => byte.toString(16))
+        .join('');
+      const salt = `0x${strSalt.padEnd(64, '0')}`;
+
+      const { rifCost, duration } = getState().registrar;
+
+      localStorage.setItem(
+        `${domain}-options`,
+        JSON.stringify({
+          salt,
+          duration,
+          rifCost,
+          contract: FIFS_ADDR_REGISTRER,
+        }),
+      );
+    }
+
+    return dispatch(receiveIsCommitmentRequired(isCommitmentRequired));
+  } catch (error) {
+    return dispatch(notifyError(error.message));
+  }
```

Comment:
We should probably have a `finally` here to revert the state that changed when the function started execution

Reply:
True

Reply:
On a second thought, reverting the state that changed when the function started execution isn't necessarily the behaviour we want, we need the state to remain the same for a while even after this function execution is over no?